pkg/aspect/common_test.go

```diff
@@ -146,3 +149,76 @@ func TestValidateTemplateExpressions(t *testing.T) {
 		})
 	}
 }
+
+type testhandler struct{ int }
+
+func (testhandler) Close() error { return nil }
+
+func TestFromHandler(t *testing.T) {
+	f := FromHandler(testhandler{37})
+	asp, err := f(nil, nil)
+	if err != nil {
+		t.Fatalf("FromHandler should never return an error")
+	}
+	if th, ok := asp.(testhandler); !ok {
+		t.Fatalf("FromHandler changed type of Handler instance")
+	} else if th.int != 37 {
+		t.Fatalf("Handler instance was messed with by FromHandler, that should never happen")
+	}
+}
+
+func TestFromBuilder(t *testing.T) {
+	tests := []struct {
+		name         string
+		kind         config.Kind
+		expectedType reflect.Type // we only compare the type of this value against the one returned by the builder
+		err          string
+		args         []interface{}
+	}{
+		{"access logs", config.AccessLogsKind, reflect.TypeOf((*adapter.AccessLogsAspect)(nil)).Elem(), "", []interface{}{}},
```

Comment:
Ah, now that I'm driving this off of Kind I can add one - I didn't have a test case for it originally because my original `switch` statement was over `builder.(type)`. I'll add one.